reconcile/utils/aws_api.py

```diff
@@ -892,6 +892,40 @@ def get_vpc_subnets(vpc_id: str, ec2: EC2Client) -> list[SubnetTypeDef]:
         subnets = ec2.describe_subnets(Filters=[{"Name": "vpc-id", "Values": [vpc_id]}])
         return subnets.get("Subnets", [])
 
+    def get_mesh_vpc_peerings(self, account, vpc_key, vpc_value: str):
+        comparing_vpc_id = ""
+        comparing_vpc_cidr = ""
+        comparing_vpc_dict = {}
+        for account_parse in account:
+            ec2 = self._account_ec2_client(account_parse["name"])
+            mesh_vpc_dict = {}
+            response = ec2.describe_vpcs()
+            for vpc in response["Vpcs"]:
+                if "Tags" in vpc:
+                    for tag in vpc["Tags"]:
+                        if tag["Key"] == vpc_key and tag["Value"] == vpc_value:
```

Comment:
`account-vpc-mesh` is just an extended version of `account-vpc`: create peering between cluster vpc and all vpcs in the target account matching all the tags. So the vpc here already contains everything we need, we don't need to do a further look into the vpc peering connection with this vpc, the rest of the peering would be covered in other clusters' validation. I suggest you reuse [get_vpcs_details](https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/aws_api.py#L944-L967) to get all vpcs , then loop each vpc and put into `peerings_enteries_dict` just like `account-vpc`